src/main/java/lithium/university/resources/TwitterResource.java

```diff
@@ -73,6 +76,21 @@ public Response getHomeTimeline() {
         return Response.ok(new Tweet(list), MediaType.APPLICATION_JSON).build();
     }
 
+    @GET
+    @Path("/tweet/filter")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getFilteredTweets(@QueryParam("keyword") Optional<String> keyword) {
+        List<TwitterPost> rawPosts = (List<TwitterPost>) ((Tweet) getHomeTimeline().getEntity()).getContent();
+
+        List<String> filteredPosts = rawPosts.parallelStream().filter(p -> p.getTwitterMessage().contains(keyword.orElse(""))).map(TwitterPost::getTwitterMessage).collect(Collectors.toList());
```

Comment:
why parallel stream and what are the pros/cons of it?

Reply:
Parallel streams allow operations with large amounts of data be processed faster because it divides the work among multiple threads. However, in this scenario, parallel stream probably isn't the best idea because:
1. There isn't a *large* amount of data (or operations taking place that require extensive amounts of time) going in 
2. Dividing the data and coordinating threads introduces a fair amount of overhead